zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Notebook.java

```diff
@@ -572,7 +573,78 @@ private long getUnixTimeLastRunParagraph(Paragraph paragraph) {
     return lastRunningUnixTime;
   }
 
-  public List<Map<String, Object>> getJobListforNotebook(boolean needsReload,
+  public List<Map<String, Object>> getJobListforNotebookFromParagraphId(String paragraphID) {
+    String gotNoteId = null;
+    List<Note> notes = getAllNotes();
+    for (Note note : notes) {
+      Paragraph p = note.getParagraph(paragraphID);
+      if (p != null) {
+        gotNoteId = note.getId();
+      }
+    }
+    return getJobListforNotebookFromNotebookId(gotNoteId);
+  }
+
+  public List<Map<String, Object>> getJobListforNotebookFromNotebookId(String notebookID) {
```

Comment:
`Note` is set of `Paragraphs`. `Notebook` is set of `Notes`.

I think this line need to be renamed, like

```
public List<Map<String, Object>> getJobListforNotebookFromNoteId(String noteId) {
```

and i think it's more simple and clear to say 

```
public List<Map<String, Object>> getJobListByNoteId(String noteId) {
```

I can see some other places `notebook` being used instead of `note`.
Could you update your changes to use `note` when it refer single note (group of paragraphs) and use `notebook` when it refer multiple notes (group of notes) ?

Let me know if you need any help.